client/landing/stepper/declarative-flow/newsletter.ts

```diff
@@ -113,10 +117,13 @@ const newsletter: Flow = {
 			switch ( _currentStep ) {
 				case 'intro':
 					if ( userIsLoggedIn ) {
-						return navigate( 'newsletterSetup' );
+						return navigate( 'newsletterType' );
 					}
 					return window.location.assign( logInUrl );
 
+				case 'newsletterType':
```

Comment:
Step name and components need a better name.

Reply:
I opted for newsletterGoals as a screen and component name.  The 'goals' terminology aligns with the longstanding Goals screen for the start flow, with how we save the options here (as site_goals), and with the language we've been using in day-to-day discussions.